### PR TITLE
Added type annotation to helpers, added ts declaration files for angu…

### PR DESCRIPTION
**Pull request by ralzinov**
## Issue:
Modules "storybook/angular" and "storybook/anguar/demo" has no type declaration files and
angular helpers has no proper typings. ts-loader throws many errors with "noExplicitAny" flag enabled. [Issue](https://github.com/storybooks/storybook/issues/2435)

## What I did
Added type declaration fles for "storybook/angular" and "storybook/anguar/demo" modules. Annotated angular helpers functions

## How to test
Set "noExplicitAny": true in tsconfig.json

Is this testable with jest or storyshots?
no

Does this need a new example in the kitchen sink apps?
no

Does this need an update to the documentation?
no

### CODE REVIEW

**Comment by danielduan** [comment]
Not sure if this is you or on master but it looks like the tests are broken:

```
WARNING in /tmp/storybook/node_modules/@angular/platform-browser-dynamic/node_modules/@angular/common/esm5/common.js
122:24-38 "export 'InjectionToken' was not found in '@angular/core'
 @ /tmp/storybook/node_modules/@angular/platform-browser-dynamic/node_modules/@angular/common/esm5/common.js
 @ /tmp/storybook/node_modules/@angular/platform-browser-dynamic/esm5/platform-browser-dynamic.js
 @ /tmp/storybook/app/angular/dist/client/preview/angular/helpers.ts
 @ /tmp/storybook/app/angular/dist/client/preview/render.js
 @ /tmp/storybook/app/angular/dist/client/preview/index.js
 @ /tmp/storybook/app/angular/dist/client/index.js
 @ ./.storybook/config.js
 @ multi /tmp/storybook/app/angular/dist/server/config/polyfills.js /tmp/storybook/app/angular/dist/server/config/globals.js /tmp/storybook/app/angular/node_modules/webpack-hot-middleware/client.js?reload=true ./.storybook/config.js

WARNING in /tmp/storybook/node_modules/@angular/platform-browser-dynamic/esm5/platform-browser-dynamic.js
123:57-89 "export 'ɵCodegenComponentFactoryResolver' was not found in '@angular/core'
 @ /tmp/storybook/node_modules/@angular/platform-browser-dynamic/esm5/platform-browser-dynamic.js
 @ /tmp/storybook/app/angular/dist/client/preview/angular/helpers.ts
 @ /tmp/storybook/app/angular/dist/client/preview/render.js
 @ /tmp/storybook/app/angular/dist/client/preview/index.js
 @ /tmp/storybook/app/angular/dist/client/index.js
 @ ./.storybook/config.js
 @ multi /tmp/storybook/app/angular/dist/server/config/polyfills.js /tmp/storybook/app/angular/dist/server/config/globals.js /tmp/storybook/app/angular/node_modules/webpack-hot-middleware/client.js?reload=true ./.storybook/config.js

WARNING in /tmp/storybook/node_modules/@angular/platform-browser-dynamic/esm5/platform-browser-dynamic.js
128:45-49 "export 'ɵcmf' was not found in '@angular/core'
 @ /tmp/storybook/node_modules/@angular/platform-browser-dynamic/esm5/platform-browser-dynamic.js
 @ /tmp/storybook/app/angular/dist/client/preview/angular/helpers.ts
 @ /tmp/storybook/app/angular/dist/client/preview/render.js
 @ /tmp/storybook/app/angular/dist/client/preview/index.js
 @ /tmp/storybook/app/angular/dist/client/index.js
 @ ./.storybook/config.js
 @ multi /tmp/storybook/app/angular/dist/server/config/polyfills.js /tmp/storybook/app/angular/dist/server/config/globals.js /tmp/storybook/app/angular/node_modules/webpack-hot-middleware/client.js?reload=true ./.storybook/config.js
```
@ndelangen who can review these angular PRs? there's a few of these just sitting around now

**Comment by ralzinov** [comment]
@danielduan I think it is some npm packages conflict in parent branch: https://stackoverflow.com/questions/44656265/export-%C9%B5cmf-was-not-found-in-angular-core

**Comment by Hypnosphi** [comment]
FYI: test passes on upstream branch: https://circleci.com/gh/storybooks/storybook/24979

**Comment by Hypnosphi** [comment]
Here it fails because of errors not warnings:
```
ERROR in /tmp/storybook/app/angular/src/client/preview/angular/helpers.ts
(52,5): error TS2322: Type 'Timer' is not assignable to type 'number'.

ERROR in /tmp/storybook/app/angular/src/client/preview/angular/helpers.ts
(151,5): error TS90010: Type 'PlatformRef' is not assignable to type 'PlatformRef'. Two different types with this name exist, but they are unrelated.

ERROR in /tmp/storybook/app/angular/dist/client/preview/angular/helpers.ts
(52,5): error TS2322: Type 'Timer' is not assignable to type 'number'.

ERROR in /tmp/storybook/app/angular/dist/client/preview/angular/helpers.ts
(151,5): error TS90010: Type 'PlatformRef' is not assignable to type 'PlatformRef'. Two different types with this name exist, but they are unrelated.
  Property '_injector' is missing in type 'PlatformRef'.
```

File: app/angular/index.d.ts
Comment (ndelangen) [inline app/angular/index.d.ts]: I think we also have type definitions in the `@types/...` npm organisation. 

Should we move these here? Or add them there also?
Are these even the same type of definitions?

👍

Comment (ralzinov) [reply app/angular/index.d.ts]: I didn't find any typings for apps, only for addons at https://microsoft.github.io/TypeSearch/. 
I think type definitioins and app should be in same package, at least written with typescript like angular. It's not the same typings.

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
@alterx do you want to approve and merge this?

**Comment by alterx** [comment]
@ndelangen lets wait until we approve and merge https://github.com/storybooks/storybook/pull/2467/files/a55741d4e1e7de20553a826172fa946d518e396e..fef3b9883089409971a028dc59db1f9b5e8b5dae

**Comment by ndelangen** [comment]
Seems like the CLI tests are breaking over this:
```
ERROR in /tmp/storybook/app/angular/src/client/preview/angular/helpers.ts
(139,5): error TS2345: Argument of type '(PipeTransform | IComponent | typeof AppComponent)[]' is not assignable to parameter of type '(any[] | Type<any>)[]'.
  Type 'PipeTransform | IComponent | typeof AppComponent' is not assignable to type 'any[] | Type<any>'.
    Type 'PipeTransform' is not assignable to type 'any[] | Type<any>'.
      Type 'PipeTransform' is not assignable to type 'Type<any>'.

ERROR in /tmp/storybook/app/angular/src/client/preview/angular/helpers.ts
(179,68): error TS2345: Argument of type '{ component: any; props: { message: string; stack: string; }; propsMeta: {}; }' is not assignable to parameter of type 'Data'.
  Property 'pipes' is missing in type '{ component: any; props: { message: string; stack: string; }; propsMeta: {}; }'.

ERROR in /tmp/storybook/app/angular/dist/client/preview/angular/helpers.ts
(139,5): error TS2345: Argument of type '(IComponent | PipeTransform | typeof AppComponent)[]' is not assignable to parameter of type '(any[] | Type<any>)[]'.
  Type 'IComponent | PipeTransform | typeof AppComponent' is not assignable to type 'any[] | Type<any>'.
    Type 'PipeTransform' is not assignable to type 'any[] | Type<any>'.
      Type 'PipeTransform' is not assignable to type 'Type<any>'.
        Property 'apply' is missing in type 'PipeTransform'.

ERROR in /tmp/storybook/app/angular/dist/client/preview/angular/helpers.ts
(179,68): error TS2345: Argument of type '{ component: any; props: { message: string; stack: string; }; propsMeta: {}; }' is not assignable to parameter of type 'Data'.
  Property 'pipes' is missing in type '{ component: any; props: { message: string; stack: string; }; propsMeta: {}; }'.
Child html-webpack-plugin for "index.html":
         Asset    Size  Chunks  Chunk Names
    index.html  569 kB       0  
       [0] /tmp/storybook/node_modules/html-webpack-plugin/lib/loader.js!/tmp/storybook/app/angular/dist/server/index.html.ejs 1.7 kB {0} [built]
       [1] /tmp/storybook/node_modules/lodash/lodash.js 540 kB {0} [built]
       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
Child html-webpack-plugin for "iframe.html":
          Asset    Size  Chunks  Chunk Names
    iframe.html  568 kB       0  
       [0] /tmp/storybook/node_modules/html-webpack-plugin/lib/loader.js!/tmp/storybook/app/angular/dist/server/iframe.html.ejs 886 bytes {0} [built]
       [1] /tmp/storybook/node_modules/lodash/lodash.js 540 kB {0} [built]
       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
error Command failed with exit code 1.
```

**Comment by ralzinov** [comment]
@ndelangen @alterx  I fixed conflicts, but i think we should enable "noImplicitAny" flag in tsconfig, because without it typescript is useless, and we will get broken typings after each merge.

**Comment by ndelangen** [comment]
What do you think @alterx @igor-dv ?

**Comment by alterx** [comment]
Yeah, I agree. If we're going to have type definitions we need to enforce its use.
On a non-related note, I think we need to fix `IGetStory` to include the new `moduleMetadata`

**Review by alterx — APPROVED** [review]

**Comment by ndelangen** [comment]
👏

**Pull request merged**
